pilot/pkg/xds/xdsgen.go

```diff
@@ -187,3 +188,37 @@ func ResourceSize(r model.Resources) int {
 	}
 	return size
 }
+
+// xdsNeedsPush checks for the common cases whether we need to push or not.
+func xdsNeedsPush(req *model.PushRequest, proxy *model.Proxy) (needsPush, definitive bool) {
+	if proxy.Type == model.Ztunnel {
+		// Not supported for ztunnel
+		return false, true
+	}
+	if req == nil {
+		return true, true
+	}
+	// If none set, we will always push
+	if len(req.ConfigsUpdated) == 0 {
```

Comment:
I think we can check if req.Full == true
```suggestion
	if req.Full == true {
```

Reply:
I think, we should keep it as is - We actually check for full and push if needed for types that only work for Full.

Reply:
>Full != configsUpdate == 0.


Yeah, Full is a subset of configsUpdate == 0.